rfcs/text/0000-extend-file-pe.md

```diff
@@ -0,0 +1,150 @@
+# 0000: Extend the PE field set
+
+- Stage: **1 (proposal)**
+- Date: **TBD**
+
+The Portable Executable (PE) sub-field, of the `file` top-level fieldset, can be updated to include more file attributes to aid in file analysis. This additional document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+This RFC is to create 25 additional sub-fields within the `file.pe` fieldset.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| pe.authentihash | keyword | Authentihash of the PE file. |
+| pe.compile_timestamp | date | Compile timestamp of the PE file. |
+| pe.compiler | nested | Compiler information. |
+| pe.compiler.version | keyword | Version of the compiler. |
+| pe.compiler.name | keyword | Name of the compiler. |
+| pe.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| pe.entry_point | keyword | Relative byte offset to the base of the PE file. |
+| pe.exports | keyword | List of symbols exported by PE |
+| pe.debug | nested | Debug information, if present |
+| pe.debug.offset | keyword | Debug offset information. |
+| pe.debug.size | keyword | Size of the debug information. |
+| pe.debug.type_str | keyword | Format of the debug information. |
```

Comment:
I'm not sure about this v. `type` -- it looks like `type_str` is just the "normalized" value of `type` with the name found in source macros. Personally I'd prefer just dropping `type_str` and storing the normalized type in `type`, but if we want maybe we can hash that out in a further step of the RFC? @ebeahan thoughts?

Reply:
Based on @andrewstucki explanation, `type` seems sufficient, but I can't speak authoritatively in this context if `type_str` is more idiomatic or accurate of what's being captured in the field.

I do agree that it doesn't have to be fully decided at this stage. It can be captured as an outstanding concern in the doc for further discussion at a later stage.

Reply:
@andrewstucki yeah, you'd mentioned this above and I'd not fixed it yet. That's done now. I did as you suggested 👍